app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -468,11 +484,11 @@ class DiscoveryView extends React.Component {
     };
     return compact([
       !projectId && {
-        label: renderTab("Projects", filteredProjectStats.count || "-"),
+        label: renderTab("Projects", filteredProjectCount || "-"),
         value: "projects",
       },
       {
-        label: renderTab("Samples", filteredSampleStats.count || "-"),
+        label: renderTab("Samples", filteredSampleCount || "-"),
         value: "samples",
       },
       domain !== DISCOVERY_DOMAIN_PUBLIC &&
```

Comment:
Seems like the counter for the Visualizations tab updates pretty inconsistently. Not sure if this is completely relevant to this PR since this is more focused on samples/projects counter, but thought I'd mention it here.

Example flow:
1) Go to All Data view. Visualizations tab should have a count.
![image](https://user-images.githubusercontent.com/53838890/74560456-7393d180-4f1b-11ea-88ca-b4682d3e0b18.png)

2) Select some filter on the left sidebar. Visualizations tab now has no count.
![image](https://user-images.githubusercontent.com/53838890/74560464-78588580-4f1b-11ea-9049-fc7850ef45b6.png)

3) Interact with the the view in some way (select a sample, toggle a sidebar, etc.) Visualizations tab now has a count again?
![image](https://user-images.githubusercontent.com/53838890/74560475-7d1d3980-4f1b-11ea-9b4b-8bde058335d3.png)

Some changes in the PR actually seem to mitigate some of this behavior, since I checked on master/staging/prod and the Visualizations counter behavior is even weirder there.

Reply:
Oh looks like QA already reported this IDSEQ-1914 😅